### PR TITLE
refactor: make resource pkg calculate pipeline task

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
make resource pkg calculate pipeline task

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   make resource pkg calculate pipeline task           |
| 🇨🇳 中文    |     抽出pipeline资源计算到pkg包         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sfwn — CHANGES_REQUESTED** [review]
maybe no need resource pkg.
just change `func (s *PipelineSvc) calculatePipelineResources` into provider.

**Review by sfwn — CHANGES_REQUESTED** [review]
merge pkg/resource into provider.

**Comment by chengjoey** [comment]
> merge pkg/resource into provider.

done

File: internal/tools/pipeline/providers/resource/provider.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+type provider struct {
+	Log       logs.Logger
+	Cfg       *config
+	ActionMgr actionmgr.Interface
+
+	bdl *bundle.Bundle
+}
+
+func (s *provider) Init(ctx servicehub.Context) error {
+	s.bdl = bundle.New()
```
Comment (sfwn) [inline internal/tools/pipeline/providers/resource/provider.go:39]: this is a bundle without any clients

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/tools/pipeline/providers/resource/provider.go (hunk 1 above)
Comment (chengjoey) [reply internal/tools/pipeline/providers/resource/provider.go]: no usage bundle, deleted

**Review by sfwn — APPROVED** [review]

**Pull request merged**